tests/integration/test_sharding.py

```diff
@@ -52,13 +52,6 @@ def test_shard_group_sanity(cluster):
     with raises(ResponseError, match='MOVED [0-9]+ 1.1.1.1:111'):
         c.set('key', 'value')
 
-    # Follower redirect straight to remote shardgroup to
```

Comment:
Why is this test no longer valid and removed?

Reply:
from slack

[Shaya Potter](https://app.slack.com/team/UM9BFGE2V)  [4:34 PM](https://redis.slack.com/archives/C027R1TTHJ6/p1648474470182639)
so this breaks a test (not the code, but a test)
[4:34](https://redis.slack.com/archives/C027R1TTHJ6/p1648474486067229)
```# Follower redirect straight to remote shardgroup to
        # avoid another redirect hop.
        cluster.wait_for_unanimity()
        cluster.node(2).wait_for_log_applied()  # to get shardgroup
        with raises(ResponseError, match='MOVED [0-9]+ 1.1.1.1:111'):
>           cluster.node(2).client.set('key', 'value')
E           AssertionError: Pattern 'MOVED [0-9]+ 1.1.1.1:111' not found in 'MOVED 0 localhost:5001'
```
[4:35](https://redis.slack.com/archives/C027R1TTHJ6/p1648474519443559)
need to explore that
[Shaya Potter](https://app.slack.com/team/UM9BFGE2V)  [4:50 PM](https://redis.slack.com/archives/C027R1TTHJ6/p1648475435841149)
previously it would be caught by handleSharding earlier and redirect to remote cluster then and never hit checkLeader() now it does.  I'm unsure if we should try to optimize it, and just remove the test?
[Ozan Tezcan](https://app.slack.com/team/U0270RYB881)  [4:55 PM](https://redis.slack.com/archives/C027R1TTHJ6/p1648475737366189)
yeah I agree, follower can just redirect to leader, leader will redirect to remote cluster anyway

Reply:
though going through the code again, might not be necessary to remove, as the handleSharding call there was put back.

Reply:
tried it with current code and it works, so putting back.